### PR TITLE
Migrate to `AddFlags` pattern.

### PR DESCRIPTION
**Pull request by mattmoor**
See discussion on this PR: https://github.com/sigstore/cosign/pull/781#discussion_r715904414

Signed-off-by: Matt Moore <mattomata@gmail.com>

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: cmd/cosign/cli/options/sign.go [hunk 1]
```diff
@@ -64,14 +64,8 @@ func AddSignOptions(cmd *cobra.Command, o *SignOptions) {
 	cmd.Flags().StringVar(&o.Attachment, "attachment", "",
 		"related image attachment to sign (sbom), default none")
 
-	AddAnnotationOptions(cmd, &o.AnnotationOptions)
+	o.AnnotationOptions.AddFlags(cmd)
 
 	cmd.Flags().BoolVar(&o.RegistryOpts.AllowInsecure, "allow-insecure-registry", false,
 		"whether to allow insecure connections to registries. Don't use this for anything but testing")
-
-	AddRekorOptions(cmd, &o.Rektor)
```
Comment (n3wscott) [inline cmd/cosign/cli/options/sign.go:72]: these get lost?

Comment (mattmoor) [reply cmd/cosign/cli/options/sign.go:72]: Must have been a bad merge.  I'd moved them to the top.  I'll add them to my push fixing the style thing.

Comment (n3wscott) [reply cmd/cosign/cli/options/sign.go:72]: there now, 👍

File: cmd/cosign/cli/options/sign.go [hunk 2]
```diff
@@ -39,8 +39,15 @@ type SignOptions struct {
...
-// AddSignOptions adds the sign command options to cmd.
-func AddSignOptions(cmd *cobra.Command, o *SignOptions) {
+var _ Interface = (*SignOptions)(nil)
+
+// AddFlags implements Interface
+func (o *SignOptions) AddFlags(cmd *cobra.Command) {
+	o.Rektor.AddFlags(cmd)
+	o.Fulcio.AddFlags(cmd)
+	o.OIDC.AddFlags(cmd)
+	o.AnnotationOptions.AddFlags(cmd)
```
Comment (n3wscott) [inline cmd/cosign/cli/options/sign.go:49]: This neeeds `AddSecurityKeyOptions` still

File: cmd/cosign/cli/options/sign.go [hunk 3]
```diff
@@ -39,8 +39,15 @@ type SignOptions struct {
 	RegistryOpts RegistryOpts
 }
 
-// AddSignOptions adds the sign command options to cmd.
-func AddSignOptions(cmd *cobra.Command, o *SignOptions) {
+var _ Interface = (*SignOptions)(nil)
+
+// AddFlags implements Interface
+func (o *SignOptions) AddFlags(cmd *cobra.Command) {
+	o.Rektor.AddFlags(cmd)
```
Comment (n3wscott) [inline cmd/cosign/cli/options/sign.go:46]:
```suggestion
        o.SecurityKey.AddFlags(cmd)
	o.Rektor.AddFlags(cmd)
```

Comment (mattmoor) [reply cmd/cosign/cli/options/sign.go:46]: added

**Review by n3wscott — APPROVED** [review]

**Pull request merged**